apps/explorer/src/pages/checkpoints/Transactions.tsx

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+
+import {
+    genTableDataFromTxData,
+    getDataOnTxDigests,
+    type TxnData,
+} from '~/components/transaction-card/TxCardUtils';
+import { useRpc } from '~/hooks/useRpc';
+import { TableCard } from '~/ui/TableCard';
+
+export function CheckpointTransactions({
+    digest,
+    transactions,
+}: {
+    digest: string;
+    transactions: string[];
+}) {
+    const rpc = useRpc();
+    const { data: txData, isLoading } = useQuery(
+        ['checkpoint-transactions', digest],
+        async () => {
+            // todo: replace this with `sui_getTransactions` call when we are
+            // able to query by checkpoint digest
+            const txData = await getDataOnTxDigests(rpc, transactions!);
+            return genTableDataFromTxData(txData as TxnData[]);
```

Comment:
Fine with this for now but we obviously want to move away from these helpers / components at some point.